### PR TITLE
solutions from 9 to 16

### PR DESCRIPTION
**Pull request by ielfurlan**

### CODE REVIEW

File: solutions/ielfurlan/s010.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+while True:
```
Comment (furlan) [inline solutions/ielfurlan/s010.py:4]: I didn't understand why you are using while loop here.

File: solutions/ielfurlan/s010.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    m = int(input('the month:'))
+    if m > 12 or m < 1:
+        print('not a valid month')
+        continue
+    m2 = m
+    break
+while True:
+    d = int(input('the day:'))
+    d2 = d
+    if m == 1 or m == 3 or m == 5 or m == 7 or m == 8 or m == 10 or m == 12:
```
Comment (furlan) [inline solutions/ielfurlan/s010.py]: I suggest you use an array or list for this. Then you can check if the value you are testing is in the array or list.

File: solutions/ielfurlan/s010.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            continue
+        d = d2 + 1
+    if d == 32:
+        d = 1
+        m = m2 + 1
+        if m == 13:
+            m = 1
+            y2 = y
+            y = y2 + 1
+    elif m == 4 or m == 6 or m == 9 or m == 11:
```
Comment (furlan) [inline solutions/ielfurlan/s010.py]: I suggest you use an array or list for this. Then you can check if the value you are testing is in the array or list.

File: solutions/ielfurlan/s010.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+while True:
+     y = int(input('the year:'))
+     if y <= -11000:
+          print('no a valid year')
+          continue
+     break
+if y % 400 == 0:
+    ly = y
```
Comment (furlan) [inline solutions/ielfurlan/s010.py]: I suggest you use leap_year instead of ly.
It's better for the clarity of the program.
One great program said some time ago: "Bad programmers write to computers, good programmers write to other programs."

File: solutions/ielfurlan/s010.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    break
+while True:
+    d = int(input('the day:'))
+    d2 = d
+    if m == 1 or m == 3 or m == 5 or m == 7 or m == 8 or m == 10 or m == 12:
+        if d > 31 or d < 1:
+            print('not a valid day')
+            continue
+        d = d2 + 1
+    if d == 32:
```
Comment (furlan) [inline solutions/ielfurlan/s010.py]: One suggestion: you could use a list or array with the last days of each month, [31,28,31,...].
You check if the entered is in the list (according to the month entered), if so, next day is 1.

File: solutions/ielfurlan/s010.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
+while True:
+     y = int(input('the year:'))
+     if y <= -11000:
```
Comment (furlan) [inline solutions/ielfurlan/s010.py:6]: Why are you using 11000?

File: solutions/ielfurlan/s011.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+lp = input('license plate (no spaces):')
+if lp[0].isalpha() and lp[1].isalpha() and lp[2].isalpha() and lp[3].isdigit() and lp[4].isdigit() and lp[5].isdigit:
```
Comment (furlan) [inline solutions/ielfurlan/s011.py:2]: You don't need to test each character; you can test first 3 or 4 characters if it's letters or numbers.
Another quick way to test the old and new license plate is test the size 6 or 7.

File: solutions/ielfurlan/s012.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from random import randint
+s = randint(0,37)
+if s == 37:
+    s = 00
+print('the spin resulted in', s, '...')
+print('pay', s)
+if s != 0 and s != 00:
+    if s == 1 or s == 3 or s == 5 or s == 7 or s == 9 or s == 12 or s == 14 or s == 16 or s == 18 or s == 19 or s == 21 or s == 23 or s == 25 or s == 27 or s == 30 or s == 32 or s == 34 or s == 36:
```
Comment (furlan) [inline solutions/ielfurlan/s012.py]: I would suggest you to use list or array, instead of several "or".

File: solutions/ielfurlan/s012.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+from random import randint
+s = randint(0,37)
+if s == 37:
+    s = 00
+print('the spin resulted in', s, '...')
+print('pay', s)
+if s != 0 and s != 00:
```
Comment (furlan) [inline solutions/ielfurlan/s012.py:8]: What happens if space is 0 or 00?

File: solutions/ielfurlan/s013.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
+b = []
```
Comment (furlan) [inline solutions/ielfurlan/s013.py:1]: You don't need to use a list and later count the number of elements to calculate the average. You can use just an integer variable and increment by 1 every number different of zero.

File: solutions/ielfurlan/s013.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+c = 0
+while True:
+    a = int(input('the number:'))
+    if a != 0:
+        b.append(a)
+        c = c + a
+    else:
+        break
+if c == 0:
```
Comment (furlan) [inline solutions/ielfurlan/s013.py:10]: Clever way to test it.

File: solutions/ielfurlan/s014.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
+a = float(input('the money input: '))
```
Comment (furlan) [inline solutions/ielfurlan/s014.py:1]: Try to use an array with the list of bills and goes through each one to find out the number of dollars bills for each.
Tip: use operators // and %=
The operators `//` and `%=` are both arithmetic operators in Python.

The `//` operator is the floor division operator. It performs division between two operands and returns the quotient rounded down to the nearest whole number (integer). It discards the decimal part of the division result. For example:

```python
a = 10
b = 3
quotient = a // b  # 3
```

In the above example, `10 // 3` results in `3` because the quotient of dividing 10 by 3 is 3.

The `%=` operator is the modulo assignment operator. It calculates the remainder when the left operand is divided by the right operand and assigns the result to the left operand. It is a shorthand way of performing the modulo operation and assigning the result back to the same variable. For example:

```python
a = 10
b = 3
a %= b  # 1
```

In the above example, `10 % 3` results in `1`, and the value of `a` is updated to `1` using the `%=` operator.

Overall, the `//` operator is used for floor division, while the `%=` operator combines the modulo operation with assignment.

File: solutions/ielfurlan/s016.py [hunk 13]
```diff
@@ -0,0 +1,6 @@
+a = input('Uppercase and lowercase matters. Put your word here: ')
+b = a[::-1]
```
Comment (furlan) [inline solutions/ielfurlan/s016.py]: Smart, but I think for the sake of learning, write another version, but not using reverse list from Python.

File: solutions/ielfurlan/s015.py [hunk 14]
```diff
@@ -0,0 +1,12 @@
+a = 3
+for b in range(1,9):
```
Comment (furlan) [inline solutions/ielfurlan/s015.py:2]: Why range 1-9? Write a comment explaining why.
BTW, there is an extensive discussion in the programming community about good comments. As a rule of thumb, good comments do not explain what your code does (your code should be clear enough), but good comments explain why you are using that logic/algorithm etc.
Remember, code is like jokes: if you need to explain, it's not good.

File: solutions/ielfurlan/s015.py
Comment (furlan) [inline solutions/ielfurlan/s015.py:1]: This solution could work, but it's hard to follow. My suggestion is to write expressions that are very similar to the "graphic" expression:

```
pi = 3 + (4 / (2 * 3 * 4)) - (4 / (3 * 4 * 5)) ...
```
I have an idea how to solve this problem, but I want you come out with your own solution first.

**Review by furlan — CHANGES_REQUESTED** [review]
I comment every file and some specifics lines. Overall, I missed some comments about your solution, proper capitalization in the user messages, and the variables names should be more meaningful. I like some solutions you submitted, and I think you are making a good progress.
Review my comments, make the corrections I suggested and let me know if you have any questions.
Good job!

File: solutions/ielfurlan/s015.py (hunk 14 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s015.py:2]: The "for in range" loop , the final count is not included and it starts at 0, unless you tell it to start at a different number.

File: solutions/ielfurlan/s012.py (hunk 9 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s012.py:8]: I forgot to add that. I will do that right now.

File: solutions/ielfurlan/s010.py (hunk 1 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s010.py:4]: to make sure that you put a valid date.

File: solutions/ielfurlan/s010.py (hunk 6 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s010.py:6]: The Universe is no more than 10000 years old.

File: solutions/ielfurlan/s011.py (hunk 7 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s011.py:2]: I need to check every character because you could do AAA11A, and it would be valid.

File: solutions/ielfurlan/s013.py (hunk 11 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s013.py:10]: kinda late, but:
thank you

File: solutions/ielfurlan/s010.py (hunk 5 above)
Comment (ielfurlan) [reply solutions/ielfurlan/s010.py]: I was gonna ignore it, but I will do it for practice.

Comment (ielfurlan) [reply solutions/ielfurlan/s010.py]: I see that I can use less lines with this.
